R/query-string.R

```diff
@@ -11,116 +11,131 @@ queryStringFilter <- function(req){
 
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == "") {
+
+  if (is.null(qs) || length(qs) == 0L || qs == "") {
     return(list())
   }
-  if (stri_startswith_fixed(qs, "?")) {
-    qs <- substr(qs, 2, nchar(qs))
-  }
 
-  parts <- strsplit(qs, "&", fixed = TRUE)[[1]]
-  kv <- strsplit(parts, "=", fixed = TRUE)
-  kv <- kv[vapply(kv, length, numeric(1)) == 2] # Ignore incompletes
+  qs <- stri_replace_first_regex(qs, "^[?]", "")
+  #qs <- stri_replace_all_fixed(qs, "+", " ")
```

Comment:
```suggestion
```

Reply:
I left it there in case we wanted to consider `+` query as a space character.